### PR TITLE
setup bulla swap

### PR DESCRIPTION
**Pull request by solidoracle**

### CODE REVIEW

File: bulla-contracts/schema.graphql [hunk 1]
```diff
@@ -286,6 +286,52 @@ type InvoiceReconciledEvent implements IEventLog & IPoolTransaction @entity {
...
+  expiry: BigInt!
+  signerWallet: Bytes! #address
+  signerToken: Token! #address
+  signerAmount: BigInt!
+  senderWallet: Bytes! #address
+  senderToken: Token! #address
+  senderAmount: BigInt!
+}
+
+interface IBullaSwapEvent {
```
Comment (bengobeil) [inline bulla-contracts/schema.graphql:301]: I'm not sure to what extent this interface is useful

File: bulla-contracts/schema.graphql [hunk 2]
```diff
@@ -286,6 +286,52 @@ type InvoiceReconciledEvent implements IEventLog & IPoolTransaction @entity {
...
+  transactionHash: Bytes!
+  logIndex: BigInt!
+  timestamp: BigInt!
+}
+
+type OrderExecutedEvent implements IEventLog & IOrderEvent @entity {
+  id: ID!
+  order: OrderERC20!
+  sender: Bytes! #address
+  recipient: Bytes! #address
```
Comment (bengobeil) [inline bulla-contracts/schema.graphql:327]: sould we keep same terminology with signerWallet?

File: bulla-contracts/src/mappings/BullaSwap.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+export function handleOrderCreated(event: OrderCreated): void {
+  const ev = event.params;
+  const orderId = ev.orderId;
+
+  // Create the order entity
+  const order = new OrderERC20(orderId.toString());
+  order.orderId = orderId;
+  order.expiry = ev.order.expiry;
+  order.signerWallet = ev.order.signerWallet;
+  order.signerToken = ev.order.signerToken.toHexString();
```
Comment (bengobeil) [inline bulla-contracts/src/mappings/BullaSwap.ts:15]: you should call getOrCreateToken(ev.order.signerToken) and pass the token here.

File: bulla-contracts/src/mappings/BullaSwap.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  order.signerAmount = ev.order.signerAmount;
+  order.senderWallet = ev.order.senderWallet;
+  order.senderToken = ev.order.senderToken.toHexString();
```
Comment (bengobeil) [inline bulla-contracts/src/mappings/BullaSwap.ts:18]: same

File: bulla-contracts/src/mappings/BullaSwap.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+export function handleOrderExecuted(event: OrderExecuted): void {
+  const ev = event.params;
+  const orderId = ev.orderId;
+
+  // Create the order entity
+  const order = new OrderERC20(orderId.toString());
+  order.orderId = orderId;
+  order.expiry = ev.order.expiry;
+  order.signerWallet = ev.order.signerWallet;
+  order.signerToken = ev.order.signerToken.toHexString();
```
Comment (bengobeil) [inline bulla-contracts/src/mappings/BullaSwap.ts:55]: same

File: bulla-contracts/src/mappings/BullaSwap.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  order.signerAmount = ev.order.signerAmount;
+  order.senderWallet = ev.order.senderWallet;
+  order.senderToken = ev.order.senderToken.toHexString();
```
Comment (bengobeil) [inline bulla-contracts/src/mappings/BullaSwap.ts:58]: same

File: bulla-contracts/tests/BullaFactoring.test.ts [hunk 7]
```diff
@@ -181,7 +181,7 @@ test("it handles BullaFactoring v2 events", () => {
 
   log.info("✅ should create a InvoiceKickbackAmountSent event with correct claim ID", []);
 
-  const totalRefundAmount = BigInt.fromI32(9000);
+  const totalRefundAmount = BigInt.fromI32(9000000);
```
Comment (bengobeil) [inline bulla-contracts/tests/BullaFactoring.test.ts:184]: hmm I would revert this. We would have never witnessed the issue if it wasn't for it.

**Review by bengobeil — APPROVED** [review]

File: bulla-contracts/schema.graphql (hunk 2 above)
Comment (solidoracle) [reply bulla-contracts/schema.graphql:327]: was thinking the same overnight

**Review by bengobeil — APPROVED** [review]

**Pull request merged**